vermouth/dssp/dssp.py

```diff
@@ -462,6 +487,73 @@ def __init__(self, attribute, sequence,
         self.molecule_selector = molecule_selector
 
     def run_molecule(self, molecule):
+        """
+        Run the processor on a single molecule.
+
+        Parameters
+        ----------
+        molecule: vermouth.molecule.Molecule
+
+        Returns
+        -------
+        vermouth.molecule.Molecule
+        """
         if self.molecule_selector(molecule):
             annotate_residues_from_sequence(molecule, self.attribute, self.sequence)
         return molecule
+
+    def run_system(self, system):
+        """
+        Run the processor on a system.
+
+        Parameters
+        ----------
+        system: vermouth.system.System
+
+        Returns
+        -------
+        vermouth.system.System
+        """
+        # Test and adjust the length of the sequence. There are 3 valid scenarios:
+        # * the length of the sequence matchesthe number of residues in the
+        #   selection;
+        # * all the molecules in the selection have the same number of residues
+        #   and the sequence length matches the number of residue of one
+        #   molecule; in this case the equence is repeated for each molecule;
+        # * the sequence has a length of one; in this case the sequence is
+        #   repeated for each residue.
+        # The case were there is no molecule in the selection is only valid if
+        # the sequence is empty. Then we are in the first valid scenario.
+        molecule_lengths = [
+            len(list(molecule.iter_residues()))
+            for molecule in system.molecules
+            if self.molecule_selector(molecule)
+        ]
+        if self.sequence and not molecule_lengths:
+            raise ValueError('There is no molecule to which '
+                             'to apply the sequence.')
+        if (molecule_lengths
+                and len(self.sequence) == molecule_lengths[0]
+                and utils.are_all_equal(molecule_lengths)):
+            sequence = list(self.sequence) * len(molecule_lengths)
```

Comment:
Can `self.sequence` be a string a this point? In that case `list(self.sequence)` will do something else than `[self.sequence]`

Reply:
`self.sequence` can be any kind of sequence. Using `list` normalizes the behaviour. is `self.sequence` is `"ABCD"` and `len(molecule_lengths)` is 2, then `self.sequence` will become `['A', 'B', 'C', 'D', 'A', 'B', 'C', 'D']` which is the expected behaviour.